source/solution_deploy/source/test/test_action_target_provider.py

```diff
@@ -0,0 +1,272 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# TODO: test that name over 20 characters is rejected
+# TODO: test that ID over 20 characters is rejected
+
+import os
+import pytest
+from pytest_mock import mocker
+import boto3
+import botocore.session
+from botocore.stub import Stubber, ANY
+from botocore.exceptions import ClientError
+import random
+from action_target_provider import (
+    lambda_handler,
+    CustomAction,
+    get_securityhub_client,
+)
+from botocore.config import Config
+
+os.environ["AWS_REGION"] = "us-east-1"
+os.environ["AWS_PARTITION"] = "aws"
+
+
+@pytest.fixture(autouse=True)
+def mock_get_account_id(mocker):
+    mocker.patch("action_target_provider.get_account_id", return_value="111111111111")
+
+
+class MockContext(object):
+    def __init__(self, name, version):
+        self.function_name = name
+        self.function_version = version
+        self.invoked_function_arn = (
+            "arn:aws:lambda:us-east-1:123456789012:function:{name}:{version}".format(
+                name=name, version=version
+            )
+        )
+        self.memory_limit_in_mb = float("inf")
+        self.log_group_name = "test-group"
+        self.log_stream_name = "test-stream"
+        self.client_context = None
+
+        self.aws_request_id = "-".join(
+            [
+                "".join([random.choice("0123456789abcdef") for _ in range(0, n)])
+                for n in [8, 4, 4, 4, 12]
+            ]
+        )
+
+
+context = MockContext("SO0111-SHARR-Custom-Action-Lambda", "v1.0.0")
+
+
+def event(type):
+    return {
+        "ResourceProperties": {
+            "Name": "Remediate with ASR Test",
+            "Description": "Test Submit the finding to Automated Security Response on AWS",
+            "Id": "ASRRemediationTest",
+        },
+        "RequestType": type,
+        "ResponseURL": "https://bogus",
+    }
+
+
+def test_get_client(mocker):
+    client1 = get_securityhub_client()
+    assert client1
+    client2 = get_securityhub_client()
+    assert client2 == client1
+
+
+def test_lambda_handler(mocker):
+    """
+    Basic check for errors
```

Comment:
This file isn't new. Unless there's a bug lurking here, I'm not inclined to change it very much at this point

Reply:
It won't be new ever again ;-) if we don't improve files while we're already looking at them, the code base will deteriorate over time. Let's avoid that.

Reply:
In principle, I agree. But you can't hold up changes that have a positive effect by requiring everyone to fix everything when they touch something. The result is that no one wants to clean up or refactor anything because it won't pass code review for the changes that do need to go in. Or we will invent ways of doing things that involve touching the least number of files to avoid comments like this. This is a balancing act, and the context of when the change is made, the impact of the cruft, and how well covered the code is by tests are critical.